dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -398,74 +391,60 @@ public Response unlockContent(@Context HttpServletRequest request, @Context Http
+		final User user = initData.getUser();
+		final String render = paramsMap.get(RESTParams.RENDER.getValue());
+		final String query = paramsMap.get(RESTParams.QUERY.getValue());
+		final String id = paramsMap.get(RESTParams.ID.getValue());
+		final String limitStr = paramsMap.get(RESTParams.LIMIT.getValue());
+		final String offsetStr = paramsMap.get(RESTParams.OFFSET.getValue());
+		final String inode = paramsMap.get(RESTParams.INODE.getValue());
+		final long language = toLong(paramsMap.get(RESTParams.LANGUAGE.getValue()),
+										() -> APILocator.getLanguageAPI().getDefaultLanguage().getId());
 		/* Limit and Offset Parameters Handling, if not passed, using default */
-
-		int limit = 10;
-		int offset = 0;
-
-		try {
-			if(UtilMethods.isSet(limitStr)) {
-				limit = Integer.parseInt(limitStr);
-			}
-		} catch(NumberFormatException e) {
-		}
-
-		try {
-			if(UtilMethods.isSet(offsetStr)) {
-				offset = Integer.parseInt(offsetStr);
-			}
-		} catch(NumberFormatException e) {
-		}
-
-		boolean live = (paramsMap.get(RESTParams.LIVE.getValue()) == null || ! "false".equals(paramsMap.get(RESTParams.LIVE.getValue())));
+		final int limit  = toInt(limitStr,  () -> 10);
+		final int offset = toInt(offsetStr, () -> 0);
+		final boolean live = (paramsMap.get(RESTParams.LIVE.getValue()) == null ||
+				!"false".equals(paramsMap.get(RESTParams.LIVE.getValue())));
 
 		/* Fetching the content using a query if passed or an id */
-
-		List<Contentlet> contentlets = new ArrayList<Contentlet>();
+		List<Contentlet> contentlets = new ArrayList<>();
 		Boolean idPassed = false;
 		Boolean inodePassed = false;
 		Boolean queryPassed = false;
+		String result = null;
+		Optional<Status> status = Optional.empty();
+		String type = paramsMap.get(RESTParams.TYPE.getValue());
+		String orderBy = paramsMap.get(RESTParams.ORDERBY.getValue());
+
+		type = UtilMethods.isSet(type)?type:"json";
+		orderBy = UtilMethods.isSet(orderBy)?orderBy:"modDate desc";
 
 		try {
+
 			if(idPassed = UtilMethods.isSet(id)) {
-				contentlets.add(this.contentHelper.hydrateContentLet(
-						APILocator.getContentletAPI().findContentletByIdentifier(id, live, language, user, true)));
+				Optional.ofNullable(this.contentHelper.hydrateContentLet(APILocator.getContentletAPI()
+						.findContentletByIdentifier(id, live, language, user, true)))
+						.ifPresent(contentlets::add);
 			} else if(inodePassed = UtilMethods.isSet(inode)) {
-				contentlets.add(this.contentHelper.hydrateContentLet(
-						APILocator.getContentletAPI().find(inode, user, true)));
+				Optional.ofNullable(this.contentHelper.hydrateContentLet(APILocator.getContentletAPI()
+						.find(inode, user, true))).ifPresent(contentlets::add);
 			} else if(queryPassed = UtilMethods.isSet(query)) {
 				String tmDate=(String)request.getSession().getAttribute("tm_date");
 				contentlets = ContentUtils.pull(query, offset, limit,orderBy,user,tmDate);
 			}
+
+			if (null == contentlets || contentlets.size() == 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)